vortex-serde/src/layouts/read/layouts/chunked.rs

```diff
@@ -0,0 +1,352 @@
+                                .stats_projection
+                                .get(&field_prefix)
+                                .vortex_expect("We got the key by iterating keys");
+                            let prefix_str = match field_prefix {
+                                Field::Name(n) => n.to_string(),
+                                Field::Index(i) => i.to_string(),
+                            };
+                            let expanded_batch = add_missing_columns(
+                                b,
+                                &prefix_str,
+                                field_stats,
+                                self.message_cache.dtype(),
+                            )?;
+                            let mask = pf.evaluate(&expanded_batch)?;
+                            let row_offsets = expanded_batch
+                                .with_dyn(|a| {
+                                    a.as_struct_array()
+                                        .ok_or_else(|| vortex_err!("Stats weren't a struct array"))
+                                        .map(|s| {
+                                            s.field_by_name(&format!("{prefix_str}_row_offset"))
+                                        })
+                                })?
+                                .ok_or_else(|| vortex_err!("Missing row offsets"))?;
+                            filter_offsets(row_offsets, mask, scan.row_count as usize)
+                                .map(|s| Some(PlanResult::Range(s)))
+                        } else {
+                            Ok(Some(PlanResult::Batch(b)))
+                        }
+                    }
+                },
+            }
+        } else {
+            let metadata_child = self
+                .flatbuffer()
+                .children()
+                .ok_or_else(|| vortex_err!("Missing children"))?
+                .get(0);
+            self.metadata_reader = Some(
+                self.layout_builder.read_layout(
+                    self.fb_bytes.clone(),
+                    metadata_child._tab.loc(),
+                    Scan {
+                        // TODO(robert): This projection is not yet utilized since stats are Flat
+                        projection: Projection::Flat(
+                            scan.stats_projection
+                                .values()
+                                .flat_map(|stats| stats.iter().map(|s| Field::Name(s.to_string())))
+                                .collect(),
+                        ),
+                        rows: None,
+                        filter: None,
+                        batch_size: usize::MAX,
+                    },
+                    self.message_cache.relative_stored_dtype(0u16),
+                )?,
+            );
+            self.plan(scan)
+        }
+    }
+}
+
+fn filter_offsets(offsets: Array, mask: Array, row_count: usize) -> VortexResult<RowSelector> {
```

Comment:
The mask has to be flipped before being passed to this function or this function flips the mask